src/main/java/TicTacToe.java

```diff
@@ -0,0 +1,327 @@
+                    if( board[i][j] == 0 )
+                        w = WinConfig.NONE ;
+                }
+        return w ;
+
+    }
+
+    private String getRowString(int row)
+    {
+        String s = "" ;
+        for( int i = 0 ; i < N ; i ++ )
+        {
+            switch(board[row][i]) {
+            case 0: s += " " ;
+                break ;
+            case 1: s += "O" ;
+                break ;
+            case 2: s += "X" ;
+            }
+
+            if( i != N-1 )
+                {
+                    s += " | " ;
+                }
+        }
+
+            s += String.format("%" + HSPACE + "s", "");
+
+         for( int i = 0 ; i < N ; i ++ )
+          {
+            s += row*3+i+1 ;
+
+            if( i == N-1 ) {
+                s += "\n";
+            }
+            else {
+                s += " | " ;
+            }
+          }
+        return s;
+    }
+
+    public String toString()
+    {
+        String s = "";
+        // iterate through the rows
+        for( int i = 0 ; i < N ; i ++ ) {
+            s += getRowString(i);
+        }
+        return s;
+    }
+
+    public static void main( String[] args )
+    {
+        System.out.println("Welcome to Tic-Tac-Toe.");
+        System.out.println("");
+
+        TicTacToe game = new TicTacToe();
+        String move_str ;
+        int move1 = 0 ;
+        int move2 = 0 ;
+        int player_type = 0 ;
+        WinConfig w = WinConfig.NONE ;
+
+        System.out.println("Please make your move selection by entering a number 1-9 corresponding to the movement key on the right.\n");
+        System.out.println(game.toString()) ;
+
+        while( game.isWinningConfig() == WinConfig.NONE  )
+        {
+            do {
+            move1 = game.getplayer1().getMove();
+            } while(!game.setMove(move1,game.getplayer1().getPlayerType() ));
+
+
+            if( ( w = game.isWinningConfig() ) == WinConfig.WIN ) {
+                System.out.println("");
+                System.out.println(game.toString() );
+                System.out.println("You have beaten my poor AI!");
+                break ;
+            }
+            else if( w == WinConfig.DRAW ) {
+                System.out.println("");
+                System.out.println(game.toString()) ;
+                System.out.println("Well played. It is a draw!");
+                break ;
+            }
+
+            move2 = game.getplayer2().getMove();
+            System.out.println("");
+            System.out.println("You have put an X in the " + TicTacToe.getPosDescription(move1) + ". I will put a O in the " + TicTacToe.getPosDescription(move2) + "." ) ;
+            game.setMove(move2, game.getplayer2().getPlayerType() ) ;
+
+            if( game.isWinningConfig() == WinConfig.WIN ) {
+                System.out.println("");
+                System.out.println(game.toString() );
+                System.out.println("I won. Thanks for playing.") ;
+                break ;
+            }

```

Comment:
We shouldn't have to repeat all this code for player 1 and player 2, this can be factored to just have one set of code to do this and manage player1 or player2